### PR TITLE
Update to go1.17

### PR DESCRIPTION
**Pull request by kevinpollet**
## What does this PR do?

This PR:

- Updates the Go version to `1.17`
- Updates the Traefik dependency version to `v2.5.6`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,24 +3,22 @@ module github.com/traefik/mesh
 go 1.16
```
Comment (ldez) [inline go.mod:3]:
```suggestion
go 1.17
```

**Review by ldez — CHANGES_REQUESTED** [review]

**Comment by ldez** [comment]
an update of golangci-lint can be useful also https://github.com/traefik/mesh/blob/d69475264c5d0200647cb626933df330900a6f59/Dockerfile#L24

**Comment by kevinpollet** [comment]
I will do the golangci-lint update thanks for pointing this!
By the way, I do not understand why the integration tests are not executed?! any ideas?

File: README.md [hunk 2]
```diff
@@ -3,7 +3,6 @@
 </p>
 
 
-[![Travis CI Build Status](https://travis-ci.com/traefik/mesh.svg?branch=master)](https://travis-ci.com/traefik/mesh)
```
Comment (tomMoulard) [inline README.md:6]: You can maybe add the GitHub action status badge instead ?

File: docs/content/contributing/building-testing.md [hunk 3]
```diff
@@ -37,7 +37,7 @@ traefik-mesh
 
 Requirements:
 
-- `Go` v1.14+
+- `Go` v1.16+
```
Comment (tomMoulard) [inline docs/content/contributing/building-testing.md:40]:
```suggestion
- `Go` v1.17+
```

Comment (tomMoulard) [reply docs/content/contributing/building-testing.md:40]: see https://go.dev/doc/modules/gomod-ref#go

**Comment by kevinpollet** [comment]
I opened another PR because the semaphore workflow is triggered.

Superseded by https://github.com/traefik/mesh/pull/809

**Pull request closed without merging**